mars/services/subtask/worker/tests/subtask_processor.py

```diff
@@ -81,6 +82,27 @@ def _execute_operand(self, ctx: Dict[str, Any], op: OperandType):
                     continue
                 self.assert_object_consistent(out, ctx[out.key])
 
+    async def _store_data(self, *args, **kwargs):
+        # `iter_output_data` must ensure values order since we only return values.
+        shuffle_output = {
+            key: data
+            for key, data, is_shuffle in iter_output_data(
+                self.subtask.chunk_graph, self._datastore
+            )
+            if is_shuffle
+        }
+        # assert output keys order consistent
+        if shuffle_output:
+            mapper_reducer_indices = defaultdict(list)
+            for chunk_key, reducer_index in shuffle_output.keys():
+                mapper_reducer_indices[chunk_key].append(reducer_index)
+            for reducer_indices in mapper_reducer_indices.values():
```

Comment:
As the assertion is a little expensive, is it possible to make it CI-only by judging from environ `CI=true`? (see usage of `_is_ci` in https://github.com/mars-project/mars/blob/master/mars/serialization/serializables/field.py).

Reply:
The assertion is added into `CheckedSubtaskProcessor`, which won't be used in production. You mean we also add the check in `mars.services.subtask.worker.processor.SubtaskProcessor` and enable it when only running in CI?

Reply:
maybe this check is not needed, if we order all mapper outputs by reducer index in ` mars.services.subtask.utils.iter_output_data`